pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -278,7 +278,7 @@ public Node getNthParent(int n) {
 
         for (int i = 0; i < node.jjtGetNumChildren(); i++) {
             Node child = node.jjtGetChild(i);
-            if (child.getClass() == targetType) {
+            if (targetType.isAssignableFrom(child.getClass())) {
```

Comment:
this change should be described in the changelog. I'll update it when merging if nothing else arises.